src/crossroads/main.py

```diff
@@ -185,8 +191,80 @@ def validate_json(json_content):
     return True
 
 
-def data_which_overlap():
-    pass
+def data_which_overlap(first_timestamp_list, second_timestamp_list):
+    """
+    Checks that the dates in both dictionary lists have an overlap.
+
+    If the lists have an overlap, returns two new lists with only the same dates.
+    If the lists do not have an overlap but the closest dates are less than 7 days apart,
+    returns two new lists that contain only those dates.
+    If the lists do not have an overlap and the closest dates are more than 7 days apart,
+    returns 'None'.
+
+    :param first_timestamp_list: (list) first user dictionaries list with coordinates
+    :param second_timestamp_list: (list) second user dictionaries list with coordinates
+    :return: (lists or None) two dictionaries lists with overlapping coordinates, or 'None' if time ranges do not match
+    """
+    first_new_list = first_timestamp_list.copy()
+    second_new_list = second_timestamp_list.copy()
+
+    first_date_from_first = datetime.fromtimestamp(int(first_timestamp_list[0]['timestamp']) / 1000)
+    last_date_from_first = datetime.fromtimestamp(int(first_timestamp_list[-1]['timestamp']) / 1000)
+    first_date_from_second = datetime.fromtimestamp(int(second_timestamp_list[0]['timestamp']) / 1000)
+    last_date_from_second = datetime.fromtimestamp(int(second_timestamp_list[-1]['timestamp']) / 1000)
```

Comment:
Is there any sense to convert timestamps to datetime objects? You can compare them with eachother with miliseconds also.